### ISSUE TITLE
Run PR tests on internal PSI resources

### ISSUE THREAD

**Issue by mohammedzee1000** [openshift/odo#4056]
We are currently battling time lost in CI due to cluster provisioning / flakyness. This is happening while we have unused resources in PSI.

I propose we start using a combination of Public Ci and internal resources to run tests, using `Message Queues` as the mediator. 
Here is a mind map based on the plan and some research

![image](https://user-images.githubusercontent.com/8547729/94664131-2a688480-0328-11eb-98b0-c023a838cc22.png)

**Comment by mohammedzee1000** [comment]
Core mechanic works as shown by POC.
As per @kadel comments, refactoring this into its own repo (https://github.com/mohammedzee1000/ci-firewall)

Procees: build and publish cli -> download requestor and worker on either side and run them

**Comment by mohammedzee1000** [comment]
Once I have it done, il update the PR and jenkins

**Comment by mohammedzee1000** [comment]
The basic piece is functional.

What is left

- [ ] Hosted rabbitmq service
- [ ] Secrets !! we cannot have full amqp url with creds and all or jenkins token hanging the repo
- [ ] Add rest of tests
- [ ] Get a fatter Jenkins (checking if we can onboard into something called Central CI which is essentially supported jenkins behind firewall)

**Comment by mohammedzee1000** [comment]
We now have a nice Red Hat IT-supported Jenkins. I have requested for a robot account as well.
Based on request by the team managing jenkins, I have modified the code to work with JMS plugin.

I have also temporarily created a CentOS 8 VPS on gcloud, using by 300$ credit, as the free rabbitmq does not provide jms support (needs jms plugin). 

Discussing with DV on how to have rabbitmq not tied to single account (like mine)

**Comment by mohammedzee1000** [comment]
Ok as far as secrets are concerned, here is what I have found out:
main docs: https://docs.ci.openshift.org/docs/architecture/step-registry/#injecting-custom-credentials
missing info in docs:

1. login to https://api.ci.openshift.org/console/
2. create a project, put your secret in it. Then open a PR to update the https://github.com/openshift/release/blob/master/core-services/secret-mirroring/_mapping.yaml file

```yaml
- from:
    namespace: your-namespace
    name: your-secret
  to:
    namespace: test-credentials
    name: your-secret
```
3. Then open another PR to inject your secret in your multistage job (as in main doc)

**Comment by mohammedzee1000** [comment]
I am currently working on setting this up and validating it

**Comment by mohammedzee1000** [comment]
Secrets are now operational

**Comment by mohammedzee1000** [comment]
Updated
![openshift pool for ci](https://user-images.githubusercontent.com/8547729/98503628-00857480-227b-11eb-8f0c-8e744cba7e94.png)
)

**Comment by mohammedzee1000** [comment]
This issue will be followed up by https://github.com/openshift/odo/issues/4328

**Comment by mohammedzee1000** [comment]
As thinks stand test flakes are becoming painful

Therefore to unblock the PR (not fixing), after discussing with @kadel, I am moving to create an optional job for now to run the poc code. see https://github.com/openshift/release/pull/15630 

Once this merges ill update the name of script as per the above PR (placeholder has already been added via https://github.com/openshift/odo/pull/4411)
